src/thunderbots/software/ai/hl/stp/evaluation/possession.cpp

```diff
@@ -44,4 +47,26 @@ namespace Evaluation
             return Evaluation::nearestRobot(team, ball.position());
         }
     }
+
+    bool teamHasPossession(const Team &team, const Ball &ball)
+    {
+        for (Robot robot : team.getAllRobots())
+        {
+            std::vector<Timestamp> robot_history_timestamps =
+                robot.getPreviousTimestamps();
+
+            int i = 0;
+
+            // Check that the robot has not had possession of the ball recently.
```

Comment:
Would this not be checking that the robot *has* had possession recently?

Reply:
Yes you are right. Should be resolved now.